test/framework/commands.go

```diff
@@ -1,13 +1,40 @@
 package framework
 
-type commandOpt func(*[]string)
+import "github.com/aws/eks-anywhere/pkg/semver"
 
-func appendOpt(new ...string) commandOpt {
-	return func(args *[]string) {
+type CommandOpt func(*string, *[]string) (err error)
+
+func appendOpt(new ...string) CommandOpt {
+	return func(binaryPath *string, args *[]string) (err error) {
 		*args = append(*args, new...)
+		return nil
 	}
 }
 
-func withKubeconfig(kubeconfigFile string) commandOpt {
+func withKubeconfig(kubeconfigFile string) CommandOpt {
 	return appendOpt("--kubeconfig", kubeconfigFile)
 }
+
+func ExecuteWithEksaVersion(version *semver.Version) CommandOpt {
+	return func(binaryPath *string, args *[]string) (err error) {
+		b, err := GetReleaseBinaryFromVersion(version)
+		*binaryPath = b
+		return err
+	}
+}
+
+func ExecuteWithLatestMinorReleaseFromVersion(version *semver.Version) CommandOpt {
+	return func(binaryPath *string, args *[]string) (err error) {
+		b, err := GetLatestMinorReleaseBinaryFromVersion(version)
+		*binaryPath = b
+		return err
+	}
+}
+
+func ExecuteWithLatestMinorReleaseFromMain() CommandOpt {
```

Comment:
No need to implement now, but maybe we could have one method that takes the branch name (or whatever other info we are gonna pass to the tests about which version of the code we are running) and decides which of this options to use?

Reply:
I'll create a sub-task for it; I'm going to take some time after this is merged and groom the stories for this area, will add this in